Sources/RxCodable/ObservableType+RxCodable.swift

```diff
@@ -17,3 +17,31 @@ public extension ObservableType where E == String {
       .map(type, using: decoder)
   }
 }
+
+public extension ObservableType where E == [String: Any] {
+  public func map<T>(_ type: T.Type, using decoder: JSONDecoder? = nil) -> Observable<T> where T: Decodable {
+    return self
+      .map { dict in try JSONSerialization.data(withJSONObject: dict) }
+      .map(type, using: decoder)
+  }
+}
+
+public extension ObservableType where E: Encodable {
+  public func toDictionary() -> Observable<[String:Any]> {
+    return self.map { encodable -> [String: Any] in
+      let data = try JSONEncoder().encode(encodable)
+      let dictionary = try JSONSerialization.jsonObject(with: data) as? [String: Any]
+      return dictionary ?? [:]
```

Comment:
This default value can cause an unexpected result. I think we should treat a casting failure as an error.

Reply:
good point!
access empty dictionary makes crash.. it would be better throw error